src/main/java/uk/gov/justice/digital/delius/jpa/standard/repository/OGRSAssessmentRepository.java

```diff
@@ -0,0 +1,22 @@
+package uk.gov.justice.digital.delius.jpa.standard.repository;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+import uk.gov.justice.digital.delius.jpa.standard.entity.OGRSAssessment;
+
+import java.util.Optional;
+
+@Repository
+public interface OGRSAssessmentRepository extends JpaRepository<OGRSAssessment, Long> {
+
+    @Query("select assessment from OGRSAssessment assessment inner join assessment.event event where event.offenderId = :offenderId and assessment.softDeleted = 0 and assessment.lastUpdatedDate = \n" +
+        "    (select max(assessmentLatest.lastUpdatedDate) \n" +
+        "    from OGRSAssessment assessmentLatest \n" +
+        "    inner join assessmentLatest.event eventLatest \n" +
+        "    where eventLatest.offenderId = :offenderId \n" +
+        "    and assessmentLatest.softDeleted = 0)")
```

Comment:
I think you might also need a event.softDeleted=0 as well. That should deal with the OGRS on an event added by mistake

Reply:
Agreed, good spot

Reply:
Also modified to do date sorting correctly to make sure we get the most recent